### PR TITLE
CLOUDP-202283: print warning for unsupported distros

### PR DESCRIPTION
**Pull request by blva**
## Proposed changes

_Jira ticket:_ [CLOUDP-202283](https://jira.mongodb.org/browse/CLOUDP-202283)

Notes:
* Another alternative, would be to run this on PersistentPreRun on the root deployment command, however, the warning would show up even if customers decide to use interactive flow (not provide flags)

### Ubuntu test
```shell
ubuntu@ip-10-128-79-112:~/atlascli$ ./bin/atlas deployments setup
? What type of deployment would you like to work with? local

[Default Settings]
Deployment Name   local305
MongoDB Version   7.0
Port              27017

? How do you want to set up your local MongoDB database? default
Local deployments are not supported on this OS, to see local deployments requirements visit https://www.mongodb.com/docs/atlas/cli/stable/atlas-cli-deploy-local/.
Error: podman not found in your system, check requirements at https://dochub.mongodb.org/core/atlas-cli-deploy-local-reqs
```

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/atlas/deployments/options/deployment_opts.go [hunk 1]
```diff
@@ -247,3 +245,48 @@ func (opts *DeploymentOpts) ValidateAndPromptDeploymentType() error {
...
+	return opts.PodmanClient.Ready(ctx)
+}
+
+func localDeploymentSupportedByOs() bool {
+	os := runtime.GOOS
+	switch os {
+	case "darwin":
+		// MacOS Intel and M1 are supported
+		return true
+	case "windows":
```
Comment (blva) [inline internal/cli/atlas/deployments/options/deployment_opts.go:263]: this should avoid that we use `sysinfo` for windows machines, but there is a test failure I'm investigating

File: internal/cli/atlas/deployments/setup.go [hunk 2]
```diff
@@ -221,10 +221,6 @@ func (opts *SetupOpts) planSteps(ctx context.Context) (steps int, needPodmanSetu
 }
 
 func (opts *SetupOpts) createLocalDeployment(ctx context.Context) error {
-	if err := podman.Installed(); err != nil {
-		return err
-	}
-
```
Comment (andreaangiolillo) [inline internal/cli/atlas/deployments/setup.go:227]: why did we remove this? Is this run as part of `opts.PodmanClient.Ready(ctx)` inside `LocalDeploymentPreRun`?

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: internal/cli/atlas/deployments/setup.go (hunk 2 above)
Comment (blva) [reply internal/cli/atlas/deployments/setup.go:227]: correct!

File: internal/cli/atlas/deployments/options/deployment_opts.go (hunk 1 above)
Comment (blva) [reply internal/cli/atlas/deployments/options/deployment_opts.go:263]: updated with a more known library

**Review by andreaangiolillo — COMMENTED** [review]
Local deployment tests are failing. It seems that is printing OS not supported for rhel 
```
 deployments_local_test.go:118:
 [2023/10/03 13:23:15.981]         	Error Trace:	/data/mci/ba99dbf9685aecc06f0fc28d64e7772e/src/github.com/mongodb/mongodb-atlas-cli/test/e2e/atlas/deployments_local_test.go:118
 [2023/10/03 13:23:15.981]         	Error:      	Not equal:
 [2023/10/03 13:23:15.981]         	            	expected: "NAME   TYPE    MDB VER   STATE\ntest   LOCAL   7.0.1     IDLE\n"
 [2023/10/03 13:23:15.981]         	            	actual  : "Local deployments are not supported on this OS, to see local deployments requirements visit https://www.mongodb.com/docs/atlas/cli/stable/atlas-cli-deploy-local/.\nNAME   TYPE    MDB VER   STATE\ntest   LOCAL   7.0.1     IDLE\n"
 [2023/10/03 13
```

https://evergreen.mongodb.com/test_log/mongodb_atlas_cli_master_e2e_required_atlas_deployments_local_e2e_patch_509511cea9def343bef1fc5ddd9a28e7254dfed7_651c134dd6d80a5c656ae92d_23_10_03_13_12_47/0?test_name=60ee52a7ef8e304493a393e2089646a8&group_id=#L8

**Review by andreaangiolillo — APPROVED** [review]

File: internal/cli/atlas/deployments/options/deployment_opts.go [hunk 3]
```diff
@@ -247,3 +245,50 @@ func (opts *DeploymentOpts) ValidateAndPromptDeploymentType() error {
 func (opts *DeploymentOpts) IsAtlasDeploymentType() bool {
 	return strings.EqualFold(opts.DeploymentType, AtlasCluster)
 }
+
+func (opts *DeploymentOpts) LocalDeploymentPreRun(ctx context.Context) error {
+	if !localDeploymentSupportedByOs() {
+		_, _ = log.Warningln("Local deployments are not supported on this OS, to see local deployments requirements visit https://www.mongodb.com/docs/atlas/cli/stable/atlas-cli-deploy-local/.")
```
Comment (blva) [inline internal/cli/atlas/deployments/options/deployment_opts.go:251]: @mongodb/docs-cloud-team asking for review here

Comment (zach-carr) [reply internal/cli/atlas/deployments/options/deployment_opts.go:251]: LGTM

**Review by zach-carr — APPROVED** [review]

**Review by andreaangiolillo — APPROVED** [review]

**Pull request merged**